jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Response.java

```diff
@@ -1272,25 +1277,7 @@ public void setTrailerFields(Supplier<Map<String, String>> trailers)
 
     protected MetaData.Response newResponseMetaData()
     {
-        return new MetaData.Response(getStatus(), getReason(), _channel.getRequest().getHttpVersion(), _fields, getLongContentLength(), getTrailers());
-    }
-
-    /**
-     * Get the MetaData.Response committed for this response.
-     * This may differ from the meta data in this response for
-     * exceptional responses (eg 4xx and 5xx responses generated
-     * by the container) and the committedMetaData should be used
-     * for logging purposes.
-     *
-     * @return The committed MetaData or a {@link #newResponseMetaData()}
-     * if not yet committed.
-     */
-    public MetaData.Response getCommittedMetaData()
-    {
-        MetaData.Response meta = _channel.getCommittedMetaData();
-        if (meta == null)
-            return newResponseMetaData();
-        return meta;
+        return new MetaData.Response(getStatus(), getReason(), _channel.getRequest().getHttpVersion(), _channel.getCoreResponse().getHeaders(), getLongContentLength(), getTrailers());
```

Comment:
Why are you using `_channel.getCoreResponse().getHeaders()` and not `_fields`?

Reply:
They should now be the same instance.... but if somehow they are not the same, then we should use the ones form the core request.  I guess I could put in:
```suggestion
        assert _channel.getCoreResponse().getHeaders() == _fields
        return new MetaData.Response(getStatus(), getReason(), _channel.getRequest().getHttpVersion(), _channel.getCoreResponse().getHeaders(), getLongContentLength(), getTrailers());
```
But currently there is no way for that to not be true, so I'm good leaving it out unless you think it makes it clearer.

Reply:
Yeah it just looks weird that all the modifications are done on the `_fields` `HttpFields.Mutable` which is passed in. But then when it is used we just rely on an external assumption that it is the one from the `getCoreResponse()`.

Do you even need `protected void onResponse(HttpFields.Mutable headers)` method when you could just assign `_fields = _channel.getCoreResponse().getHeaders()`?